optimize/nlls/lm_test.go

```diff
@@ -0,0 +1,242 @@
+package lm
```

Comment:
please add a license header.

also: the package is named `lm` while the directory holding the code is `nlls`.
this is a bit confusing. they should somewhat match.

Reply:
Changed to nlls.